src/decawave/src/decawave_node.cpp

```diff
@@ -50,54 +49,30 @@ void update_callback(const ros::TimerEvent&);
 
 // ROS Params
 double frequency = 50.0;
-int port_num = 0;
+std::string port_name= "ttyACM0";
 // for(int i = 0; i < argc; ++i){
 // int port_num = atoi(argv[1]);
 // }
-
 // Global_Vars
 Decawave *piTag;
-decawave_coordinate tagPos;
-// Decawave piTag(1);
-// decawave_coordinate tagPos;
 
 int main(int argc, char** argv){
-  // Init ROS
-  // ros::init(argc, argv, "decawave_node");
-  // nh = new ros::NodeHandle("");
-  // pnh = new ros::NodeHandle("~");
-
   // Params
-  // pnh->param<double>("frequency", frequency);
-  // nh->param<int>("port_num",port_num);
-  // port_num = nh->getParam("port_num",port_num);
-  // int port_num_test = ros::param::get("~port_num",port_num);
-  ros::param::get("~port_num",port_num);
-  // debug:
-  // std::string s = std::to_string(port_num);
-  // char const *pchar = s.c_str();  //use char const* as target type
-  // ROS_INFO(pchar);
-  printf("%i\n",port_num);
-  // printf("%i\n",port_num_test);
-  //nh->param<param_name3_type>(param_name3_path, param_name3, param_name3_default;
-  ros::init(argc, argv, "decawave_node" + std::to_string(port_num));
+  ros::param::get("~port_name",port_name);
+
+  // Init this ROS node
+  ros::init(argc, argv, port_name);
```

Comment:
No, the name of the node should be "decawave_node", not port_name. 
There are a few features in ROS that are legacy features not practically used. 
The name here has to be unique, but it gets changed, anyway, in the launch file, when you give it a name. That's why it goes first.